disco_aws_automation/asiaq_cli.py

```diff
@@ -0,0 +1,160 @@
+"""
+Code for a conventional entry-point based command-line interface.
+"""
+
+import argparse
+import os
+from logging import getLogger
+
+import boto3
+
+from . import DiscoVPC, DiscoAWS, read_config
+from .disco_aws_util import read_pipeline_file, graceful
+from .disco_logging import configure_logging
+
+
+class CliCommand(object):
+    """
+    Abstract(ish) base class for CLI subcommand drivers.  Common behaviors of all subcommands
+    can be implemented here, but of course there's no real guarantee that they'll be honored.
+
+    Required elements in subclasses:
+
+    * a static field called DESCRIPTION, which will be used in arg parsers
+    * a class or static method called "init_args", which takes an ArgumentParser as its
+      input and adds to it whatever arguments and options this command needs.
+    """
+
+    DESCRIPTION = "This command has no description.  Perhaps you should add one?"
+
+    def __init__(self):
+        self.logger = getLogger(type(self).__name__)
+
+    def run(self, args):
+        raise Exception("This is an abstact method. Override it so this command does something useful!")
+
+    @classmethod
+    def init_args(cls, parser):
+        """
+        Set up any arguments and options for this command/subcommand.
+        """
+        pass
+
+
+class SandboxCommand(CliCommand):
+    """
+    Command to manage sandboxes (currently, just creates, but should get other functions later).
+    """
+
+    DESCRIPTION = "Create and populate a sandbox for local development and testing."
+
+    PEERING_SECTION = 'peerings'
+    BLACK_LIST_S3_CONTAINER = "zookeeper-sync-black-lists"
+    BLACK_LIST_BUCKET = 'us-west-2.mclass.sandboxes'
```

Comment:
I ask because we don't have such a layer, other than the project configuration, and that seems like a bad place to park this.  So, right now we're saying we need a different layer to hold basically the ZK-blacklist stuff and the word "sandbox" as an envtype, and much as I am in favor of appropriate layering and isolation, this sets off a little bit of an Architecture Astronauts alarm (ironic, this) in my head.